hnn_core/network.py

```diff
@@ -1374,12 +1374,17 @@ def clear_connectivity(self):
 
     def clear_drives(self):
         """Remove all drives defined in Network.connectivity"""
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] not in self.external_drives.keys():
-                connectivity.append(conn)
+        self.connectivity = [conn for conn in self.connectivity if
+                             conn['src_type'] if conn['src_type'] not
+                             in self.external_drives.keys()]
+
```

Comment:
Why is there a double if statement? Are there cases where a connection does not have `src_type`? If so,  the `if conn['src_type']` will throw a key error and not False. It would be better to use the `.get()` method, to test if a key is in a dict, which will return a `None` if the key does not exist. 
```

Reply:
oops that's a typo! interesting that it works :smile: